src/Log/LogService.php

```diff
@@ -0,0 +1,209 @@
+
+    public static function getInstance()
+    {
+        if (null === static::$instance) {
+            static::$instance = new static();
+        }
+        return static::$instance;
+    }
+
+    public function openLog($logFileName = null)		//change to createLog('default', logFileName)
+    {
+        $this->getConfig();
+
+        $logfilePath = $this->logFilePath == null ? $this->setPath($logFileName) : $this->logFilePath;
+
+        if ($logFileName == null) {
+        	$this->logFileName = 'opus.log'
+        } else {
+        	$this->logFileName = $logFileName;
+        }
+        $this->logFile = @fopen($logfilePath, 'a', false);
+        if ($this->logFile === false) {
+            $path = dirname($logfilePath);
+
+            if (! is_dir($path)) {
+                throw new Exception('Directory for logging does not exist');
+            } else {
+                throw new Exception('Failed to open logging file:' . $logfilePath);
+            }
+        }
+    }
+
+    public function getConfig()		//DONE not going to work
+    {
+        $this->config = \Zend_Registry::get('Zend_Config');
+    }
+
+    public function getPath()
+    {
+        return $this->logFilePath;
+    }
+
+    public function setPath($logFileName, $logPath = null)
+    {
+        if ($logPath == null) {
+            $this->logFilePath  = $this->config->workspacePath . '/log/' . $logFileName;
+        } else {
+            $this->logFilePath = $logPath . $logFileName;
+        }
+        return $this->logFilePath;
+    }
+
+    public function getDefaultLog()
+    {
+        if (array_key_exists('default.log', $this->loggers)) {
+            return $logger = $this->loggers['default.log'];
+        } else {
+            return $this->setDefaultLog('default.log');
+        }
+    }
+
+    public function setDefaultLog($logFileName)
+    {
+        $this->openLog('Zend_Config', $logFileName);
+        return $this->addLog();
+    }
+
+    public function addLog($logger = null)		//Done change to addLog (shouldn't be necessary)
+    {
+        if ($logger == null) {
+            $logger = $this->createLog();
+        }
+
+        if ($this->checkLogLevel() == false) {
+            $logLevelName = 'INFO';
+            $logger->warn('Log level not configured, using \'' . $logLevelName . '\'.');
+        } else {
+            $logLevelName = $this->checkLogLevel();
+        }
+
+        $zendLogRefl = new \ReflectionClass('Zend_Log');
+
+        $invalidLogLevel = false;
+
+        $logLevel = $zendLogRefl->getConstant($logLevelName);
+
+        if (empty($logLevel)) {
+            $logLevel = Zend_Log::INFO;
+            $invalidLogLevel = true;
+        }
+
+        // filter log output
+        $priorityFilter = new \Zend_Log_Filter_Priority($logLevel);
+        \Zend_Registry::set('LOG_LEVEL', $logLevel);
+        $logger->addFilter($priorityFilter);
+
+        if ($invalidLogLevel) {
+            $logger->err('Invalid log level \'' . $logLevelName .
```

Comment:
I assume this is from the old code. However since we want to improve it, try using "Invalid log level '$logLevelName' ..." (and so on). We normally use single quotes, they are safer. However when constructing strings like this double quotes and using the variable names inside the string make the code more readable.